### ISSUE TITLE
Media view is too slow when swiping pictures

### ISSUE THREAD

**Issue by motatuc** [signalapp/Signal-Android#7355]
I have:
- [x] searched open and closed issues for duplicates
- [x] read https://github.com/WhisperSystems/Signal-Android/wiki/Submitting-useful-bug-reports

----------------------------------------

### Bug description
The swipe gesture in the media view is way too slow and inconsistent in terms of loading the next picture. When swiping it sometimes takes a few seconds to load/preview the next picture. For some pictures it's really fast, for some it takes long and sometimes (similar to #7332) it scrolls really slow from one picture to the other.

### Steps to reproduce
- use the media view to show a picture
- start scrolling
- see that it takes a long time to load the next one

**Actual result:** It sometimes takes long to load the next picture or the transition between two pictures is not as intended.
**Expected result:** Scrolling pictures should be smooth and fast (as it is in the gallery).

### Device info
**Android version:** 7.1.2
**Signal version:** 4.15.3
**Memory available:** 2 GB RAM
**CPU**: Quadcore

### Link to debug log
Would a debug-log be helpful?

### Additional comment
Maybe related to: #7208 and #7332

**Comment by 2-4601** [comment]
> Would a debug-log be helpful?

Yes.

**Comment by dcherian** [comment]
I have this issue on a Moto G4 Plus.

A similar trouble I have is that after tapping on images in a conversation, it can take a really long time for the full-screen image to show up. I noticed that this started to happen after the glide update a while ago.

In the log below, I've swiped through a bunch of images (some switched quickly, others took some time to switch). I have also opened individual conversations and tapped on images to load them. It looks like the recent images generally load quickly while older ones can take 5-10 sec to load.

https://gist.githubusercontent.com/anonymous/8072fff996b745af7be6c2d6a8776937/raw/9484f6b46a5e56b868d9e019d6e2530a43f282f3/cat.log
